Refactor Userstate management

WarlockJa opened the pull request:
I have replaced roomUsers object in socket.js with UsersState to contain username data. On front-end modified RoomCard.tsx and /frontend/src/app/chat/[roomId]/page.tsx to work with the new UserState.

diwash007 reviewed (commented):
Thank you so much for your contribution, @WarlockJa 
I see a lot of TODOs on the comments .. are these all done or yet to be done?
If already completed, please remove the redundant comments

WarlockJa commented:
Hello! Removed unnecessary comments.

diwash007 commented:
Thank you so much @WarlockJa 
I will review your commits tomorrow 
Have a great day!

diwash007 commented on backend/routes/socket.js:
```diff
@@ -13,22 +13,123 @@ const io = new Server(server, {
+    socket.emit(
+      "receive_message",
+      buildMsg(
+        `You have joined the ${roomId === "1" ? "Global" : roomId} chat room`,
```
<img width="480" alt="image" src="https://github.com/diwash007/kurakani/assets/39879720/d2e10972-1296-467c-9a75-56d347a6e9fa">

diwash007 replied:
while joining private room

diwash007 commented on backend/routes/socket.js:
```diff
@@ -13,22 +13,123 @@ const io = new Server(server, {
+      .to(roomId)
+      .emit(
+        "receive_message",
+        buildMsg(`${user.username} has joined the room`, roomId)
```
Can you please change the join and leave message to the following format?
`<username>` has joined
`<username>` has left
no need to specify the room name. + no need of "You" as well

diwash007 commented on backend/routes/socket.js:
```diff
@@ -13,22 +13,123 @@ const io = new Server(server, {
 
-let roomUsers = {};
+// adding server-side Users state
+const UsersState = {
```
can we please organize the state and functions into different specific files?
That way it will be more organized to checkout the code.

WarlockJa commented:
Hello! Removed notifications to the joining user. Shortened notification for joined/left users as described. Moved UserState and its function to a separate userState.js file.

diwash007 commented:
awesome work,
One last thing, Can you please make your commit title more descriptive?

WarlockJa commented:
Sure. What do you have in mind?
